tests/integration/test_load_and_serve_agent.py

```diff
@@ -0,0 +1,134 @@
+            capture_output=True,
+            check=True,
+        )
+        return True if result.returncode == 0 else False  # noqa: TRY300
+    except Exception:
+        return False
+
+
+@pytest.mark.skipif(
+    os.environ.get("ANY_AGENT_INTEGRATION_TESTS", "FALSE").upper() != "TRUE",
+    reason="Integration tests require `ANY_AGENT_INTEGRATION_TESTS=TRUE` env var",
+)
+@pytest.mark.asyncio
+async def test_load_and_serve_agent(agent_framework: AgentFramework, tmp_path: Path) -> None:
+    kwargs = {}
+
+    tmp_file = "tmp.txt"
+
+    if not check_uvx_installed():
+        msg = "uvx is not installed. Please install it to run this test."
+        raise RuntimeError(msg)
+
+    def write_file(text: str) -> None:
+        """write the text to a file in the tmp_path directory
+
+        Args:
+            text (str): The text to write to the file.
+
+        Returns:
+            None
+        """
+        with open(os.path.join(tmp_path, tmp_file), "w", encoding="utf-8") as f:
+            f.write(text)
+
+    kwargs["model_id"] = "gpt-4.1-mini"
+    env_check = validate_environment(kwargs["model_id"])
+    if not env_check["keys_in_environment"]:
+        pytest.skip(f"{env_check['missing_keys']} needed for {agent_framework}")
+
+    model_args: dict[str, Any] = (
+        {"parallel_tool_calls": False}
+        if agent_framework not in [AgentFramework.AGNO, AgentFramework.LLAMA_INDEX]
+        else {}
+    )
+    model_args["temperature"] = 0.0
+    tools = [
+        write_file,
+        MCPStdio(
+            command="uvx",
+            args=["mcp-server-time", "--local-timezone=America/New_York"],
+            tools=[
+                "get_current_time",
+            ],
+        ),
+    ]
+    agent_config = AgentConfig(
+        tools=tools,  # type: ignore[arg-type]
+        instructions="Search the web to answer",
+        model_args=model_args,
+        **kwargs,  # type: ignore[arg-type]
+    )
+    agent_server = await AnyAgent.create_async(agent_framework, agent_config, tracing=TracingConfig())
+    logger.info(f"Agent created: {agent_server}")
+    await agent_server.serve(
+        serving_config=ServingConfig(port=5555,endpoint="/test_agent")
+    )
+    logger.info(f"Agent serving")
+
+    await asyncio.sleep(300)
+
+    # TODO use an agent card instead
+
+    # open another call in another thread
+    client = A2AClient(url = "http://localhost:5555/test_agent")
+    await client.send_task(
+        {"id": "1",
+        "message": Message(
+            role='user',
+            parts=[TextPart(text="Use the tools to find what year it is in the America/New_York timezone and write the value (single number) to a file")])
+        }
+        )
+
+    assert os.path.exists(os.path.join(tmp_path, tmp_file))
+    with open(os.path.join(tmp_path, tmp_file)) as f:
+        content = f.read()
+    assert content == str(datetime.now().year)
+    assert isinstance(agent_trace, AgentTrace)
```

Comment:
Where does agent_trace come from?

Reply:
It should come from the result of the client query. But since I cannot make the client send the query to the server yet, I'm happy with it :D

Reply:
IIRC we're not passing the whole trace as the a2a answer. If we in fact do, I'll retrieve that from the agent result and check. If we don't, I'll just test the results from the agent. I'd make a custom otel endpoint to gather the spans on a memory object and do the checks on it.